doc/cookbook/type/create_field_type_extension.rst

```diff
@@ -0,0 +1,220 @@
+.. index::
+   single: Type; Field type extension
+
+How to Create a Field Type Extension
+====================================
+
+:doc:`Custom search field types <create_custom_field_type>` are great when
```

Comment:
No problem ;) It was huge job getting this all done :scream: so I'm glad you can help with the review.